api/konnect/v1alpha1/konnect_gateway_controlplane_types.go

```diff
@@ -80,11 +80,18 @@ type KonnectGatewayControlPlaneSpec struct {
 
 // MirrorSpec contains the Konnect Mirror configuration.
 type MirrorSpec struct {
+	// Konnect contains the KonnectID of the KonnectGatewayControlPlane that
+	// is mirrored.
+	Konnect MirrorKonnect `json:"konnect"`
+}
+
+// MirrorKonnect contains the Konnect Mirror configuration.
+type MirrorKonnect struct {
 	// KonnectID is the ID of the Konnect entity. It can be set only in case
 	// the ControlPlane type is Mirror.
 	//
 	// +kubebuilder:validation:Required
-	KonnectID commonv1alpha1.KonnectIDType `json:"konnectID"`
+	ID commonv1alpha1.KonnectIDType `json:"ID"`
```

Comment:
Across this repository we've been using uncapitalized `id` for the json tag.

e.g.

- https://github.com/Kong/kubernetes-configuration/blob/7dde593f190e9516679ab8fdd7b5d9c56a1b3817/api/konnect/v1alpha1/konnect_entity_status.go#L8
- https://github.com/Kong/kubernetes-configuration/blob/ce54839705f97145b4298e7dbb65985613dc2254/api/konnect/v1alpha1/konnect_cloudgateway_dataplane_configuration_types.go#L218
- https://github.com/Kong/kubernetes-configuration/blob/d58aedc42407d610fd1edb3162d4c95da2fbf7f5/api/configuration/v1alpha1/kongcredential_jwt_types.go#L73

I'd say lets stay consistent with that

```suggestion
	ID commonv1alpha1.KonnectIDType `json:"id"`
```

Reply:
yep, I think `ID` capitalized adheres better to the naming best practices, but since we are already using `id`, I've adapted to that in 54f01f5bc53b4919f8017fda41d63d36cba33204